app/controllers/AccountSummaryController.scala

```diff
@@ -37,28 +38,29 @@ trait AccountSummaryController extends AtedBaseController with AtedFrontendAuthH
   def dataCacheConnector: DataCacheConnector
 
 
-  def view() = AuthAction(AtedRegime) {
-    implicit atedContext =>
+  def view(): Action[AnyContent] = Action.async { implicit request =>
+    authorisedAction { implicit authContext =>
       for {
         _ <- dataCacheConnector.clearCache()
         allReturns <- summaryReturnsService.getSummaryReturns
-        _ <- detailsService.cacheClientReference(atedContext.user.atedReferenceNumber)
+        _ <- detailsService.cacheClientReference(authContext.atedReferenceNumber.get)
```

Comment:
Red - a .get on an optional value could give us some None.get errors in future.

Reply:
Removed and refactored to use String as opposed to Option String